ts/features/bonus/bpd/screens/iban/insertion/IbanInsertionComponent.tsx

```diff
@@ -73,23 +74,25 @@ export const IbanInsertionComponent: React.FunctionComponent<Props> = props => {
             </Body>
             <View spacer={true} large={true} />
             <H5>{ibanDescription}</H5>
-            <Item error={isInvalidIban}>
-              <Input
-                value={iban}
-                style={IBANInputStyle}
-                autoCapitalize={"characters"}
-                maxLength={IbanMaxLength}
-                onChangeText={text => {
+            <LabelledItem
+              type="text"
+              isValid={!iban ? undefined : isValidIban}
```

Comment:
It may be worth to explore a refactoring where `iban: Option<string>` to better express the empty string meaning: `iban.fold(undefined, Iban.decode) // either undefined, right (isValid), left (isNotValid)`

Reply:
@pietro909 I tried a solution, let me know if it's ok, sorry, I'm still not very knowledgeable about fp-ts

Reply:
>  sorry, I'm still not very knowledgeable about fp-ts

Never be sorry for that :-) this time is you, the next time it will be me being reviewed by you and I'm sure I will have something to learn as well.